tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -172,7 +172,7 @@ static DetachedSpan detachInternal(String operation, SpanType type) {
 
     private static Optional<String> getParentSpanId(@Nullable Trace trace) {
         if (trace != null) {
-            Optional<OpenSpan> maybeOpenSpan = trace.pop();
+            Optional<OpenSpan> maybeOpenSpan = trace.top();
```

Comment:
are there tests to catch this sort of thing?

Reply:
yep